docs/StardustDocs/topics/predicateJoin.md

```diff
@@ -0,0 +1,283 @@
+[//]: # (title: predicateJoin)
+
+<!---IMPORT org.jetbrains.kotlinx.dataframe.samples.api.Join-->
+
+Joins two [`DataFrames`](DataFrame.md) by a join expression. 
+
+```kotlin
+predicateJoin(otherDf, type = JoinType.Inner) [ { joinExpression } ]
+
+joinExpression: JoinedDataRow.(JoinedDataRow) -> Boolean
+
+interface JoinedDataRow: LeftDataRow {
+    
+    val right: RightDataRow
+    
+}
+```
+
+Unlike [join](join.md) similar to one from relational databases, where relations between tables are based on IDs and usually equality between values, predicate join allows any operations that returns Boolean value. That gives you the ability to join data based on various logical operations and heuristics. This can also be useful when dealing with data from disparate sources that are connected, but lack common identifiers or structured relations typical of SQL tables with indices.
+
+For example, you can match rows based on:
+* **Order relations** such as `>`, `<`, `in` for numerical or DateTime values
+* **Numerical relations**
+* **Spatial relationships**, like distance within a certain range if your data includes spatial or geographical values
+* **String equivalence** using more complex comparison techniques, such as regular expressions, Levenshtein Distance.
+
+### Join types
+
+Supported join types:
+* `Inner` (default) — only matched rows from left and right [`DataFrames`](DataFrame.md)
+* `Filter` — only matched rows from left [`DataFrame`](DataFrame.md)
+* `Left` — all rows from left [`DataFrame`](DataFrame.md), mismatches from right [`DataFrame`](DataFrame.md) filled with `null`
+* `Right` — all rows from right [`DataFrame`](DataFrame.md), mismatches from left [`DataFrame`](DataFrame.md) filled with `null`
+* `Full` — all rows from left and right [`DataFrames`](DataFrame.md), any mismatches filled with `null`
+* `Exclude` — only mismatched rows from left
+
+For every join type there is a shortcut operation:
+
+#### Inner join
+
+<!---FUN predicateJoin-->
+<tabs>
+<tab title="Properties">
+
+```kotlin
+campaigns.predicateJoin(visits) {
+    right.date in startDate..endDate
+}
+```
+
+</tab>
+<tab title="Accessors">
+
+```kotlin
+val date by column<LocalDate>()
+val startDate by column<LocalDate>()
+val endDate by column<LocalDate>()
+
+campaigns.predicateJoin(visits) {
+    right[date] in startDate()..endDate()
+}
+```
+
+</tab>
+<tab title="Strings">
+
+```kotlin
+campaigns.predicateJoin(visits) {
+    right[{ "date"<LocalDate>() }] in "startDate"<LocalDate>().."endDate"<LocalDate>()
```

Comment:
is this the only notation possible to get the typed column/cell from `right`? Looks cool, but overcomplicated.

Reply:
Another notation is `right.getValue<LocalDate>("date")`. I was trying to make it shorter because it didn't fit in single line, but now samples are split in 5 different cells, so no longer an issue. I'll return `right.getValue<LocalDate>("date")`